configure.ac

```diff
@@ -231,6 +231,12 @@ LIBMESH_TEST_CXX11_NOEXCEPT
 #-----------------------------------------------------
 LT_INIT([disable-static])
 
+# We need to set the no-undefined flag for linking 
+# with libtool, because Windows requires all shared
+# libraries to include all used symbols. This flag
+# should do no harm on other platforms.
+libmesh_LDFLAGS+="-no-undefined"
```

Comment:
Here's the full error message:

<pre>
configure:21001: checking for libMesh linkage
configure:21021: mpic++ -o conftest -O0 -felide-constructors -g -pedantic -W -Wall -Wextra -Wno-long-long -Wunused -Wpointer-arith -Wformat -Wparentheses -Woverloaded-virtual -fopenmp -g -O2 -I/femputer/pbauman/civet_build_testing_root/libmesh-dbg-install/include -I/femputer/pbauman/swstack/libs/hdf5/1.8.16/gcc/6.2.0/include -I/femputer/pbauman/swstack/libs/vtk-dbg/7.1.0/gcc/6.2.0/mpich/3.1.4/include/vtk-7.1 -pthread -I/femputer/pbauman/swstack/libs/petsc/3.7.4/gcc/6.2.0/mpich/3.1.4/openblas/0.2.18/opt/include -I/femputer/pbauman/swstack/libs/petsc/3.7.4/gcc/6.2.0/mpich/3.1.4/openblas/0.2.18/opt//include -I/femputer/pbauman/swstack/libs/mpich/3.1.4/gcc/6.2.0/include -I/femputer/pbauman/swstack/libs/boost-dbg/1.61.0/gcc/6.2.0/include -DDEBUG -D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_PEDANTIC  -no-undefined  conftest.cpp -L/femputer/pbauman/swstack/libs/hdf5/1.8.16/gcc/6.2.0/lib -lhdf5 -lhdf5_cpp -Wl,-rpath,/femputer/pbauman/swstack/libs/hdf5/1.8.16/gcc/6.2.0/lib -Wl,-rpath,/femputer/pbauman/swstack/libs/vtk-dbg/7.1.0/gcc/6.2.0/mpich/3.1.4/lib -L/femputer/pbauman/swstack/libs/vtk-dbg/7.1.0/gcc/6.2.0/mpich/3.1.4/lib -lvtkIOCore-7.1 -lvtkCommonCore-7.1 -lvtkCommonDataModel-7.1 -lvtkFiltersCore-7.1 -lvtkIOXML-7.1 -lvtkImagingCore-7.1 -lvtkIOImage-7.1 -lvtkImagingMath-7.1 -lvtkIOParallelXML-7.1 -lvtkParallelMPI-7.1 -lvtkParallelCore-7.1 -lz -L/femputer/pbauman/swstack/libs/petsc/3.7.4/gcc/6.2.0/mpich/3.1.4/openblas/0.2.18/opt/lib -lpetsc -Wl,-rpath,/femputer/pbauman/swstack/libs/petsc/3.7.4/gcc/6.2.0/mpich/3.1.4/openblas/0.2.18/opt/lib -lsuperlu_dist -lcmumps -ldmumps -lsmumps -lzmumps -lmumps_common -lpord -lparmetis -lmetis -lscalapack -Wl,-rpath,/femputer/pbauman/swstack/libs/petsc/3.7.4/gcc/6.2.0/mpich/3.1.4/openblas/0.2.18/opt/lib64 -L/femputer/pbauman/swstack/libs/petsc/3.7.4/gcc/6.2.0/mpich/3.1.4/openblas/0.2.18/opt/lib64 -lsuperlu -lHYPRE -L/femputer/pbauman/swstack/libs/mpich/3.1.4/gcc/6.2.0/lib -L/femputer/pbauman/swstack/apps/gcc/6.2.0/lib/gcc/x86_64-pc-linux-gnu/6.2.0 -L/femputer/pbauman/swstack/apps/gcc/6.2.0/lib64 -L/femputer/pbauman/swstack/apps/gcc/6.2.0/lib -Wl,-rpath,/femputer/pbauman/swstack/libs/openblas/0.2.18/gcc/6.2.0/lib -L/femputer/pbauman/swstack/libs/openblas/0.2.18/gcc/6.2.0/lib -lopenblas -lmpifort -lgfortran -lquadmath -lm -lmpicxx -lstdc++ -Wl,-rpath,/femputer/pbauman/swstack/apps/gcc/6.2.0/lib/gcc/x86_64-pc-linux-gnu/6.2.0 -Wl,-rpath,/femputer/pbauman/swstack/apps/gcc/6.2.0/lib64 -Wl,-rpath,/femputer/pbauman/swstack/apps/gcc/6.2.0/lib -Wl,-rpath,/femputer/pbauman/swstack/libs/mpich/3.1.4/gcc/6.2.0/lib -lmpi -lgcc_s -ldl -Wl,-rpath,/femputer/pbauman/civet_build_testing_root/libmesh-dbg-install/lib -L/femputer/pbauman/civet_build_testing_root/libmesh-dbg-install/lib -lmesh_dbg  >&5
g++: error: unrecognized command line option '-no-undefined'; did you mean '-Wno-undef'?
</pre>

Hand-built GCC 6.2.0 on CentOS 6.